remoteworkitem/flatten_test.go

```diff
@@ -0,0 +1,20 @@
+package remoteworkitem
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestFlattenMap(t *testing.T) {
+	testString := []byte(`{"name":"shoubhik","assignee":{"1":"sbose","2":"pranav","participants":{"4":"sbose56","5":"sbose78"}},"name":"shoubhik"}`)
+	var nestedMap map[string]interface{}
+	err := json.Unmarshal(testString, &nestedMap)
+
+	if err != nil {
+		t.Error("Incorrect dataset ", testString)
+	}
+
+	OneLevelMap := Flatten(nestedMap)
+	t.Log("Initial multi level map : ", nestedMap)
+	t.Log("Flattened map : ", OneLevelMap)
```

Comment:
Yes Aslak, the code is incomplete.

I would be accessing the keys which we would be needing as part of the
concluding lines in the test case.

Also, the code logic in the DFS has a minor fix - will be pushing it
shortly.

On Sep 8, 2016 5:16 PM, "Aslak Knutsen" <notifications@github.com> wrote:

> In remoteworkitem/flatten_test.go
> <https://github.com/almighty/almighty-core/pull/227#discussion_r77990651>:
>
> > +	"encoding/json"
> > +	"testing"
> > +)
> > +
> > +func TestFlattenMap(t *testing.T) {
> > +	testString := []byte(`{"name":"shoubhik","assignee":{"1":"sbose","2":"pranav","participants":{"4":"sbose56","5":"sbose78"}},"name":"shoubhik"}`)
> > +	var nestedMap map[string]interface{}
> > +	err := json.Unmarshal(testString, &nestedMap)
> > +
> > +	if err != nil {
> > +		t.Error("Incorrect dataset ", testString)
> > +	}
> > +
> > +	OneLevelMap := Flatten(nestedMap)
> > +	t.Log("Initial multi level map : ", nestedMap)
> > +	t.Log("Flattened map : ", OneLevelMap)
>
> Logging to console and 'oh boy, the program completed without a core dump
> on me' is not a good test case.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/almighty/almighty-core/pull/227/files/69936461f2cdabf864066b0417c58f939010f3e9#r77990651>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAhSAG00WZhV70WxpD1hn0Yzpxq34D-pks5qn_V4gaJpZM4J3vMU>
> .
>